### PR TITLE
Fix doubleclick's usage of UseSameDomainRenderingUntilDeprecated

### PR DESCRIPTION
**Pull request by bradfrizzell**
doubleclick's a4a config has been completely ignoring the usage of useSameDomainRenderingUntilDeprecated

Up till now, this has not been noticed because things still work properly if a publisher is specifying USDRUD and is also using a custom remote.html.

### CODE REVIEW

File: extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js [hunk 1]
```diff
@@ -131,8 +132,14 @@ export class DoubleclickA4aEligibility {
           'be supported starting on March 29, 2018. Please refer to ' +
           'https://github.com/ampproject/amphtml/issues/11834 ' +
           'for more information');
+    let json;
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js]: const usdrd = 'useSameDomainRenderingUntilDeprecated';
const hasUSDRD = (tryParseJson(element.getAttribute('json')) || {})[usdrd] || usdrd in element.dataset;
if (hasUSDRD) {
   warnDeprecation(usrdr);
}

Comment (bradfrizzell) [reply extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js]: done

File: examples/doubleclick-usdrud.html [hunk 2]
```diff
@@ -0,0 +1,73 @@
+<!doctype html>
+<html ⚡>
+<head>
+  <meta charset="utf-8">
+  <title>Doubleclick Example with UseSameDomainRenderingUntilDeprecated</title>
+  <link rel="canonical" href="http://nonblocking.io/" >
+  <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+  <style amp-custom>
```
Comment (keithwrightbos) [inline examples/doubleclick-usdrud.html]: I don't think any of the CSS is necessary right?

File: extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js [hunk 3]
```diff
@@ -132,7 +133,8 @@ export class DoubleclickA4aEligibility {
           'https://github.com/ampproject/amphtml/issues/11834 ' +
           'for more information');
     const usdrd = 'useSameDomainRenderingUntilDeprecated';
-    const hasUSDRD = usdrd in element.dataset || element.hasAttribute(usdrd);
+    const hasUSDRD = (tryParseJson(element.getAttribute('json')) || {})[usdrd]
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js]: Actually let's reverse the items in this or as usdrd in element.dataset is faster to execute

File: extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-a4a-config.js [hunk 4]
```diff
@@ -135,15 +135,34 @@ describe('doubleclick-a4a-config', () => {
...
-          expect(elem.getAttribute(EXPERIMENT_ATTRIBUTE)).to.not.be.ok;
-        });
+    /**
+     * UseSameDomainRenderingUntilDeprecated is a flag that publishers can
+     * specify that will force them out of Fast Fetch, and force that GPT is
+     * used, not Glade. There have been many issues with this flag not being
+     * correctly honored in the past. This test checks multiple different
+     * ways that this test could be specified to assure they all work.
+     */
+    it('should use DF if UseSameDomainRenderingUntilDeprecated in use', () => {
```
Comment (keithwrightbos) [inline extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-a4a-config.js]: Nit: UseSameDomainRenderingUntilDeprecated -> useSameDomainRenderingUntilDeprecated

File: extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js (hunk 3 above)
Comment (bradfrizzell) [reply extensions/amp-ad-network-doubleclick-impl/0.1/doubleclick-a4a-config.js]: done

File: extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-a4a-config.js (hunk 4 above)
Comment (bradfrizzell) [reply extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-a4a-config.js]: done

File: examples/doubleclick-usdrud.html (hunk 2 above)
Comment (bradfrizzell) [reply examples/doubleclick-usdrud.html]: removed

File: examples/doubleclick-usdrud.html [hunk 5]
```diff
@@ -0,0 +1,34 @@
+<!doctype html>
+<html ⚡>
+<head>
+  <meta charset="utf-8">
+  <title>Doubleclick Example with UseSameDomainRenderingUntilDeprecated</title>
+  <link rel="canonical" href="http://nonblocking.io/" >
+  <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+  <script async custom-element="amp-ad" src="https://cdn.ampproject.org/v0/amp-ad-0.1.js"></script>
+  <script async custom-element="amp-analytics" src="https://cdn.ampproject.org/v0/amp-analytics-0.1.js"></script>
```
Comment (keithwrightbos) [inline examples/doubleclick-usdrud.html]: Sorry similarly these other extensions are not needed.

**Pull request merged**